changelog.d/2921.enhancement.rst

```diff
@@ -0,0 +1 @@
+JSON files are now stored without indentation.
```

Comment:
Maybe also just add why this was done, i.e. lower disk usage and better performance.

Also, the filename should be "enhance" rather than "enhancement"